src/ui/components/TestSuite/suspense/TestEventDetailsCache.ts

```diff
@@ -171,6 +184,83 @@ export const testEventDomNodeCache: ExternallyManagedCache<
   },
 });
 
+interface FindDOMNodeObjectIdForPersistentIdResult {
+  persistentId: string;
+  domNodeId: string;
+}
+
+const domNodeIdForPersistentIdCache: Cache<
+  [replayClient: ReplayClientInterface, point: ExecutionPoint, persistentIds: (string | number)[]],
+  { pauseId: string; domNodeIds: FindDOMNodeObjectIdForPersistentIdResult[] }
+> = createCache({
+  debugLabel: "DomNodeIdForPersistentIdCache",
+  getKey: ([replayClient, point, persistentIds]) => `${point}:${persistentIds.join()}`,
+  async load([replayClient, point, persistentIds]) {
+    if (findDOMNodeObjectIdForPersistentIdExpression === null) {
+      await lazyImportCypressExpression();
+    }
+
+    const evalResults: RunEvaluationResult[] = [];
+
+    // The runtime expects these to be numbers, so we need to convert them.
+    const persistentIdsAsNumbers = persistentIds.map(id => Number(id));
+
+    const sources = await sourcesByIdCache.readAsync(replayClient);
+
+    await replayClient.runEvaluation(
```

Comment:
Any particular reason why you're not using a regular evaluation here?

Reply:
Agreed. I think it might be easier to just use the `pauseEvaluationsCache` for this (with a Pause id for the `beforeStep` execution point)

Reply:
The Playwright logic below uses `runEvaluation`, so I was being consistent.

Thinking about it, the Playwright code relies on `fullPropertyPreview: true` to get additional data being returned, but in this case I'm serializing the data as a string, so a plain evaluation would work okay.